src/GitHub.InlineReviews/Peek/InlineCommentPeekableItemSource.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using GitHub.Extensions;
+using GitHub.Factories;
+using GitHub.InlineReviews.Commands;
+using GitHub.InlineReviews.Services;
+using GitHub.InlineReviews.ViewModels;
+using GitHub.Services;
+using Microsoft.VisualStudio.Language.Intellisense;
+
+namespace GitHub.InlineReviews.Peek
+{
+    class InlineCommentPeekableItemSource : IPeekableItemSource
+    {
+        readonly IApiClientFactory apiClientFactory;
+        readonly IInlineCommentPeekService peekService;
+        readonly IPullRequestSessionManager sessionManager;
+        readonly INextInlineCommentCommand nextCommentCommand;
+        readonly IPreviousInlineCommentCommand previousCommentCommand;
+
+        public InlineCommentPeekableItemSource(
+            IApiClientFactory apiClientFactory,
+            IInlineCommentPeekService peekService,
+            IPullRequestSessionManager sessionManager,
+            INextInlineCommentCommand nextCommentCommand,
+            IPreviousInlineCommentCommand previousCommentCommand)
+        {
+            this.apiClientFactory = apiClientFactory;
+            this.peekService = peekService;
+            this.sessionManager = sessionManager;
+            this.nextCommentCommand = nextCommentCommand;
+            this.previousCommentCommand = previousCommentCommand;
+        }
+
+        public void AugmentPeekSession(IPeekSession session, IList<IPeekableItem> peekableItems)
+        {
+            if (session.RelationshipName == InlineCommentPeekRelationship.Instance.Name)
+            {
+                var viewModel = new InlineCommentPeekViewModel(
+                    apiClientFactory,
+                    peekService,
+                    session,
+                    sessionManager,
+                    nextCommentCommand,
+                    previousCommentCommand);
+                viewModel.Initialize().Forget();
+                peekableItems.Add(new InlineCommentPeekableItem(viewModel));
```

Comment:
The viewmodel may not be finished initializing by the time it's added to this list. What is the impact of that?